release-lockfile.spec.md

```diff
@@ -276,14 +286,148 @@ this project depends on.
     * Values **must** be valid IPFS URIs which resolve to a valid *Release Lock File*
 
 
-## Object Definitions
+## Definitions
 
 Definitions for different objects used within the release lockfile.  All
 objects allow custom fields to be included.  Custom fields **should** be
 prefixed with `x-` to prevent name collisions with future versions of the
 specification.
 
 
+### The *Link Reference* Object
+
+A link reference object has the following key/value pairs.  All link references
+are assumed to be associated with some corresponding bytecode.
+
+#### Offset: `offset`
+
+The `offset` field is an integer which defines the start position for the link
+reference.  The location is a 0-indexed offset from the beginning of the binary
+representation of the corresponding bytecode.  This field is invalid if it
+references a position that is beyond the end of the bytecode.
+
+* Required: Yes
+* Type: Integer
+
+
+#### Length: `length`
+
+The `length` field is an integer which defines the length in bytes of the link
+reference.  This field is invalid if the end of the defined link reference
+exceeds the end of the bytecode.
+
+* Required: Yes
+* Type: Integer
+
+
+#### Name: `name`
+
+The `name` field is a string which **must** be a valid *Contract Name*.  Any
+link references which **should** be linked with the same *link value*
+**should** be given the same name.
+
+* Required: No
+* Type: String
+* Format: **must** conform to the *Contract Name* format.
+
+
+### The *Link Value* Object
+
+A *Link Value* object is defined to have the following key/value pairs.
+
+
+#### Offset `offset`
+
+The `offset` field defines the location within the corresponding bytecode where
+the `value` for this *link value* was written.  This location is a
+0-indexed offset from the beginning of the binary representation of the
+corresponding bytecode.
+
+* Required: Yes
+* Type: Integer
+* Format: The integer **must** conform to all of the following:
+    * be greater than or equal to zero 
+    * strictly less than the length of the unprefixed hexidecimal representation of the corresponding bytecode.
```

Comment:
Suggestion: rename this to `offsets`, convert to array of integers.

The same link value can and often is used to fill in multiple gaps in the same bytecode. Since this already constitutes a breaking change, I propose generalizing this field for convenience.

Reply:
(It may make sense to do the same for LinkReference)